datahub-web-react/src/app/entity/glossaryTerm/profile/RelatedEntitiesTitle.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { useGetSearchResultsForMultipleQuery } from '../../../../graphql/search.generated';
+import { SearchAcrossEntitiesInput } from '../../../../types.generated';
+import { useEntityData } from '../../shared/EntityContext';
+import { UnionType } from '../../../search/utils/constants';
+import { generateOrFilters } from '../../../search/utils/generateOrFilters';
+
+export const RelatedEntitiesTitle = () => {
+    const [relatedEntitiesCount, setRelatedEntitiesCount] = useState<number | undefined>(0);
+
+    const { entityData } = useEntityData();
+
+    const entityUrn = entityData?.urn;
+
+    const fixedOrFilters =
+        (entityUrn && [
+            {
+                field: 'glossaryTerms',
+                values: [entityUrn],
+            },
+            {
+                field: 'fieldGlossaryTerms',
+                values: [entityUrn],
+            },
+        ]) ||
+        [];
+    entityData?.isAChildren?.relationships.forEach((term) => {
+        const childUrn = term.entity?.urn;
+
+        if (childUrn) {
+            fixedOrFilters.push({
+                field: 'glossaryTerms',
+                values: [childUrn],
+            });
+
+            fixedOrFilters.push({
+                field: 'fieldGlossaryTerms',
+                values: [childUrn],
+            });
+        }
```

Comment:
this logic should not be duplicated.. could you extract it into shared logic from where you got it so it is reused?